contracts/multisig/src/key.rs

```diff
@@ -405,8 +409,32 @@ mod ecdsa_tests {
         let signature: Signature = (KeyType::Ecdsa, invalid_signature).try_into().unwrap();
         let message = MsgToSign::try_from(ecdsa_test_data::message()).unwrap();
         let public_key = PublicKey::try_from((KeyType::Ecdsa, ecdsa_test_data::pub_key())).unwrap();
-        let result = signature.verify(&message, &public_key).unwrap();
-        assert_eq!(result, false);
+        let result = signature.verify(&message, &public_key);
+        assert_eq!(
+            result.unwrap_err(),
+            ContractError::SignatureVerificationFailed {
+                reason: "unable to verify signature".into(),
+            }
```

Comment:
I'd would like to see the reason to be replaced with an error_stack context switch, but that's probably beyond the scope of this PR

Reply:
agree, I thought the same, but I preferred to leave it for another PR